core/admin/admin.php

```diff
@@ -701,6 +701,14 @@ public function admin_action_new_post() {
 
 		$post_data = Utils::get_super_global_value( $_GET, 'post_data' ) ?? [];
 
+		if ( ! current_user_can( 'publish_posts' ) && isset( $post_data['post_status'] ) && $post_data['post_status'] === 'publish' ) {
+			$post_data['post_status'] = 'draft';
+		}
+
+		$allowed_post_data_keys = [ 'post_title', 'post_content', 'post_excerpt', 'post_status', 'post_type' ];
+		$post_data = array_intersect_key( $post_data, array_flip( $allowed_post_data_keys ) );
+		$post_data = array_map( 'sanitize_text_field', $post_data );
```

Comment:
708 - 710 doesn't look clear - let's move to a separate function with self-descriptive name. Also let's add a unit test.